extensions/sql-database-projects/src/common/utils.ts

```diff
@@ -228,3 +228,26 @@ export async function getSqlProjectFilesInFolder(folderPath: string): Promise<st
 
 	return results;
 }
+
+/**
+ * Gets the Uris of all the SQL projects in the workspace
+ */
+export function getSqlProjectsInWorkspace(): vscode.Uri[] {
+	return vscode.workspace.workspaceFile ? getWorkspaceProjects().map(project => toUri(project)).filter(p => path.parse(p.fsPath).ext === constants.sqlprojExtension) : [];
+}
+
+/**
+ * Gets all the files in dataworkspace.projects
+ */
+function getWorkspaceProjects(): string[] {
```

Comment:
ideally, project provider extension shouldn't know such implementation detail, and it is subject to change as well, I'd recommend to expose these features you need as data-workspace extension's exports and you can call it here. all these features have been implemented in data-workspace. 

at this step, you could safely assume that the data-workspace extension is loaded.

Reply:
ok that makes sense to not have that in the sql projects extension. I updated the data-workspace extension's activate to return the WorkspaceService so that sql projects can get the projects in the workspace. Is this the way you were suggesting or is there a better way to do this?

Reply:
let's define an interface for the stuff we want to expose to extensions and add to it if needed, for now you only need the getProjectsInWorkspace.